pkg/cmd/pipelinerun/describe_test.go

```diff
@@ -1212,3 +1212,122 @@ func TestPipelineRunDescribeV1beta1_taskrun_with_no_status(t *testing.T) {
+				tb.TaskRunStartTime(clock.Now().Add(2*time.Minute)),
+				cb.TaskRunCompletionTime(clock.Now().Add(5*time.Minute)),
+				tb.StatusCondition(apis.Condition{
+					Type:   apis.ConditionSucceeded,
+					Status: corev1.ConditionTrue,
+				}),
+			),
+		),
+		tb.TaskRun("tr-2", "ns",
+			tb.TaskRunStatus(
+				tb.TaskRunStartTime(clock.Now().Add(5*time.Minute)),
+				cb.TaskRunCompletionTime(clock.Now().Add(9*time.Minute)),
+				tb.StatusCondition(apis.Condition{
+					Type:   apis.ConditionSucceeded,
+					Status: corev1.ConditionTrue,
+				}),
+			),
+		),
+		tb.TaskRun("tr-3", "ns",
+			tb.TaskRunStatus(
+				tb.TaskRunStartTime(clock.Now().Add(5*time.Minute)),
+				cb.TaskRunCompletionTime(clock.Now().Add(9*time.Minute)),
+				tb.StatusCondition(apis.Condition{
+					Type:   apis.ConditionSucceeded,
+					Status: corev1.ConditionTrue,
+				}),
+			),
+		),
+	}
+
+	/*
+		pipelinerun.GetAllPipelineRuns( ) returns a slice of strings containing all pipelineruns starting from the most recent one,
+		hence here the recent one is kept as the first element of the slice `pipelineRuns`
+	*/
+	pipelineRuns := []*v1alpha1.PipelineRun{
+		tb.PipelineRun("pipeline-run", "ns",
+			cb.PipelineRunCreationTimestamp(clock.Now()),
+			tb.PipelineRunLabel("tekton.dev/pipeline", "pipeline"),
+			tb.PipelineRunSpec("pipeline"),
+			tb.PipelineRunStatus(
+				tb.PipelineRunTaskRunsStatus("tr-1", &v1alpha1.PipelineRunTaskRunStatus{
+					PipelineTaskName: "t-1",
+					Status:           &trs[0].Status,
+				}),
+				tb.PipelineRunTaskRunsStatus("tr-2", &v1alpha1.PipelineRunTaskRunStatus{
+					PipelineTaskName: "t-2",
+					Status:           &trs[1].Status,
+				}),
+				tb.PipelineRunStatusCondition(apis.Condition{
+					Status: corev1.ConditionTrue,
+					Reason: resources.ReasonSucceeded,
+				}),
+				tb.PipelineRunStartTime(clock.Now()),
+				cb.PipelineRunCompletionTime(clock.Now().Add(15*time.Minute)),
+			),
+		),
+		tb.PipelineRun("pipeline-run2", "ns",
+			cb.PipelineRunCreationTimestamp(clock.Now().Add(5*time.Minute)),
+			tb.PipelineRunLabel("tekton.dev/pipeline", "pipeline"),
+			tb.PipelineRunSpec("pipeline"),
+			tb.PipelineRunStatus(
+				tb.PipelineRunTaskRunsStatus("tr-1", &v1alpha1.PipelineRunTaskRunStatus{
+					PipelineTaskName: "t-1",
+					Status:           &trs[0].Status,
+				}),
+				tb.PipelineRunTaskRunsStatus("tr-3", &v1alpha1.PipelineRunTaskRunStatus{
+					PipelineTaskName: "t-3",
+					Status:           &trs[1].Status,
+				}),
+				tb.PipelineRunStatusCondition(apis.Condition{
+					Status: corev1.ConditionTrue,
+					Reason: resources.ReasonSucceeded,
+				}),
+				tb.PipelineRunStartTime(clock.Now()),
+				cb.PipelineRunCompletionTime(clock.Now().Add(15*time.Minute)),
+			),
+		),
+	}
+
+	namespaces := []*corev1.Namespace{
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "ns",
+			},
+		},
+	}
+
+	version := "v1alpha1"
+	tdc := testDynamic.Options{}
+	dynamic, err := tdc.Client(
+		cb.UnstructuredPR(pipelineRuns[0], version),
```

Comment:
To include all the test data, wouldn't you want the second PipelineRun as well? Not familiar with using the dynamic client with tests. /cc @pradeepitm12

Reply:
Yaa, I think I should add pipelineRuns[1] as well, my bad.

Reply:
Yes, resources should be added in the dynamic client, but it is not necessary to do add all, the resources which are accessed(get, list, watch, delete, etc) with the help of the dynamic client will give return nil if the resource is not added in the test-dynamic client.
Hope this helps, for any other thing please do let me know.Yes, resources should be added in the dynamic client, but it is not necessary to do add all, the resources which are accessed(get, list, watch, delete, etc) with the help of the dynamic client will give return nil if the resource is not added in the test-dynamic client.
Hope this helps, for any other thing please do let me know.

Reply:
Done, already added.